lib/uploader/index.js

```diff
@@ -29,7 +29,20 @@ export function upload (args) {
       converter.on('error', err => reject(err))
 
       storage = createS3StorageStream(type, id, {userId, fileType, filename})
-      storage.on('finish', () => resolve(storage.url))
+      storage.on('finish', () => {
+        let attachmentType = fileType && fileType.mime.split('/')[0]
+
+        attachmentType = ['file', 'image'].includes(attachmentType)
```

Comment:
sort of an odd way to add a default

Reply:
Hmm. I'm curious what implementation you'd find less odd that'd achieve the logic required logic here and otherwise better express it? 

---
That was the TL;DR. Here is the long answer:

Lines 33-37 are expressing preeminently a transformation from mimetype to a valid Media.type which reverts to a default Media.type on the final line. I suspect that the oddness you're sensing here is symptomatic of the way Media.type is implemented as a meta category over mimetype more than the way I've implemented the logic, but maybe you're pointing to something more rudimentary. Either way I would say that the valid Media.types and it's default should probably be their own constants (VALID_MEDIA_TYPES and DEFAULT_MEDIA_TYPE, respectively) and in the best world those constants should be shared between the front and backend as they function as related delineations for both backend conversations as well as display logic. I think we're within relatively easy reach from this branch forward to migrating Media.type to Media.mimetype and this should all get a freshen up along with that...

Reply:
nevermind, it was a smaller thing but actually I think this is fine